apps/builder/app/builder/features/style-panel/sections/filter/filter-content.tsx

```diff
@@ -0,0 +1,97 @@
+import type { InvalidValue, LayersValue } from "@webstudio-is/css-engine";
+import {
+  Flex,
+  theme,
+  Label,
+  Tooltip,
+  Text,
+  TextArea,
+  textVariants,
+} from "@webstudio-is/design-system";
+import { InfoCircleIcon } from "@webstudio-is/icons";
+import { useState } from "react";
+import type { IntermediateStyleValue } from "../../shared/css-value-input";
+import { parseFilter } from "@webstudio-is/css-data";
+
+type FilterContentProps = {
+  index: number;
+  filter: string;
+  onEditLayer: (index: number, layers: LayersValue) => void;
+};
+
+export const FilterSectionContent = ({
+  index,
+  filter,
+  onEditLayer,
+}: FilterContentProps) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    IntermediateStyleValue | InvalidValue | undefined
+  >();
+
+  const handleChange = (value: string) => {
+    setIntermediateValue({
+      type: "intermediate",
+      value,
+    });
+  };
+
+  const handleComplete = () => {
+    if (intermediateValue === undefined) {
+      return;
+    }
+
+    const layers = parseFilter(intermediateValue.value);
+    if (layers.type === "invalid") {
+      setIntermediateValue({
+        type: "invalid",
+        value: intermediateValue.value,
+      });
+      return;
+    }
+
+    onEditLayer(index, layers);
+  };
+
+  return (
+    <Flex
+      direction="column"
+      css={{
+        px: theme.spacing[9],
+        paddingTop: theme.spacing[5],
+        paddingBottom: theme.spacing[9],
+        gap: theme.spacing[3],
+        minWidth: theme.spacing[30],
+      }}
+    >
+      <Label>
+        <Flex align={"center"} gap={1}>
+          Code
+          <Tooltip
+            variant="wrapped"
+            content={<Text>TODO: Add content for the filter tooltip</Text>}
```

Comment:
@todo

Reply:
Should i leave it as a todo, or can you point me a place where i can get some content to fill in here that would be great.